decidim-assemblies/app/models/decidim/assembly.rb

```diff
@@ -54,5 +60,25 @@ def hashtag
     def to_param
       slug
     end
+
+    def ancestors
+      self.class.where("#{self.class.table_name}.parents_path @> ? AND #{self.class.table_name}.id != ?", parents_path, id)
+    end
+
+    private
+
+    # rubocop:disable Rails/SkipsModelValidations
+    def set_parents_path
```

Comment:
@rbngzlv can you explain the paths concept? I don't understand it

Reply:
@oriolgual @mrcasals asked me to have the `parents_ids` cached in a column, so we can render the breadcumbs faster without a 1+N query. I have implemented it using the `ltree` extension of PostgreSQL. 

This allow us to update all descendants (child of child, ...) of an assembly in a single query when their parent changes and also to save the "path" for the actual record only querying their direct parent.

You can check this blog post for more information: http://patshaughnessy.net/2017/12/13/saving-a-tree-in-postgres-using-ltree

Reply:
Maybe it would be nice to add some docs explaining this then, since it's not quite common.